androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -81,13 +85,28 @@ class AgendaDetailsViewModel(
             is AgendaDetailsScreenEvent.OnAttendeeTabChange -> _screenStateFlow.update { it.copy(curTab = event.newTab) }
             is AgendaDetailsScreenEvent.OnEndDateChange -> updateEndDate(event.newDate)
             is AgendaDetailsScreenEvent.OnEndTimeChange -> updateEndTime(event.newHour, event.newMinute)
-            AgendaDetailsScreenEvent.OnAddPhotoClick -> TODO()
+            is AgendaDetailsScreenEvent.OnAddPhoto -> addPhoto(event.uri)
             is AgendaDetailsScreenEvent.OnPhotoClick -> _screenStateFlow.update { it.copy(enlargedPhoto = event.photo) }
             AgendaDetailsScreenEvent.CloseLargePhoto -> _screenStateFlow.update { it.copy(enlargedPhoto = null) }
             is AgendaDetailsScreenEvent.OnPhotoDelete -> deletePhoto(key = event.key)
         }
     }
 
+    @OptIn(ExperimentalUuidApi::class)
+    private fun addPhoto(uri: Uri) {
+        val agendaItem = screenStateFlow.value.agendaItem
+
+        if (agendaItem !is Event) {
+            return
+        }
+
+        val newPhotos = agendaItem.photos.toMutableList()
+        // TODO: check if uri.toString() is work
+        newPhotos.add(Photo(key = Uuid.random().toString(), url = uri.toString()))
+
+        _screenStateFlow.update { it.copy(agendaItem = agendaItem.copy(photos = newPhotos)) }
```

Comment:
Instead of working with a mutable list just to add an item, you can also just it it to a read-only list:
```
it.copy(
    agendaItem = agendaItem.copy(
        photos = agendaItem.photos + newPhoto
    )
)
```